### PR TITLE
Use SQLCipher to encrypt sqlite database

### PR DESCRIPTION
**Pull request by KhalilBellakrid**
SQLCipher is deactivated for Windows: https://ledgerhq.atlassian.net/browse/LLC-100

### CODE REVIEW

File: core/lib/cmake/ProjectSQLiteCipher.cmake [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        set(OS_COMPILER "iPhoneOS")
+        set(_host_ "arm-apple-darwin")
+        set(SKIP_BUILD_SQLCIPHER ON)
+    endif()
+
+    set(CROSS_TOP "${XCODE_DEVELOPER_DIR}/Platforms/${OS_COMPILER}.platform/Developer")
+    set(CROSS_SDK "${OS_COMPILER}.sdk")
+
+    set(_configure_options_ ${_configure_options_} --host=${_host_})
+    set(_c_flags_ "${_c_flags_} -arch ${CMAKE_OSX_ARCHITECTURES} -isysroot ${CROSS_TOP}/SDKs/${CROSS_SDK}")
```
Comment (phaazon) [inline core/lib/cmake/ProjectSQLiteCipher.cmake]: Are you sure this is right way to set C flags?

File: core/src/database/SQLite3Backend.cpp [hunk 2]
```diff
@@ -75,8 +75,11 @@ ledger::core::SQLite3Backend::setService(const std::string &service) {
 }
 
 void ledger::core::SQLite3Backend::init(const std::shared_ptr<ledger::core::api::PathResolver> &resolver,
-                                        const std::string &dbName, soci::session &session) {
-    session.open(*soci::factory_sqlite3(), resolver->resolveDatabasePath(dbName));
+                                        const std::string &dbName,
+                                        const std::string &password,
+                                        soci::session &session) {
+    auto paramaters = fmt::format("dbname=\"{}\" ", resolver->resolveDatabasePath(dbName)) + fmt::format("key=\"{}\" ", password);
```
Comment (phaazon) [inline core/src/database/SQLite3Backend.cpp]: Typo: parameters

File: tools/build_android/android_build.sh [hunk 3]
```diff
@@ -13,14 +13,14 @@ export JAVA_HOME="$(/usr/libexec/java_home -v 1.8)" || export JAVA_HOME="/usr/li
...
 
 export ARCH=$1
-if [ "${ARCH}" == 'armeabi-v7a' ]; then
-	export TOOLCHAIN_NAME='android-ndk-r16b-api-21-armeabi-v7a-clang-libcxx14'
-elif [ "${ARCH}" == 'arm64-v8a' ]; then
-	export TOOLCHAIN_NAME='android-ndk-r16b-api-24-arm64-v8a-clang-libcxx14'
-else
-	export TOOLCHAIN_NAME='android-ndk-r16b-api-16-x86-clang-libcxx14'
+if [ "$ARCH" == "armeabi-v7a" ]; then
+    export TOOLCHAIN_NAME='android-ndk-r16b-api-21-armeabi-v7a-clang-libcxx14'
```
Comment (phaazon) [inline tools/build_android/android_build.sh]: Are those tabs? 😢

File: core/lib/cmake/ProjectSQLiteCipher.cmake (hunk 1 above)
Comment (KhalilBellakrid) [reply core/lib/cmake/ProjectSQLiteCipher.cmake]: Yes I'm sure about it, do you know a better way to do so ?

File: CMakeLists.txt [hunk 4]
```diff
@@ -22,6 +22,19 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
 
 set(CMAKE_MACOSX_RPATH 1)
 
+# SQLCIPHER is used by default
+# TODO: activate SQLCipher for Windows
```
[the remark this answers is not in the document]
Comment (KhalilBellakrid) [reply CMakeLists.txt]: Yes, first let's see if Andrii can solve it easily

File: .circleci/build_lib.sh [hunk 5]
```diff
@@ -149,5 +153,5 @@ if [ "$1" == "ios" ]; then
     echo " >>> Starting iOS build for architecture ${ARCH} with toolchain ${TOOLCHAIN_NAME} for ${OSX_SYSROOT}"
     xcodebuild -project ledger-core.xcodeproj -configuration Release -jobs 4
 else
-    make -j4
+    make -j8
```
[the remark this answers is not in the document]
Comment (KhalilBellakrid) [reply .circleci/build_lib.sh]: Indeed, it's better to be dynamic !

File: core/src/database/SQLite3Backend.cpp (hunk 2 above)
Comment (KhalilBellakrid) [reply core/src/database/SQLite3Backend.cpp]: My bad ! Good catch though !

File: tools/build_android/android_build.sh (hunk 3 above)
Comment (KhalilBellakrid) [reply tools/build_android/android_build.sh]: :trollface: I'll fix it

Comment (KhalilBellakrid) [reply tools/build_android/android_build.sh]: Actually I've just checked it seems to be spaces 🤔

**Comment by KhalilBellakrid** [comment]
I push forced because of this merge : https://github.com/LedgerHQ/lib-ledger-core/pull/82

File: CMakeLists.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (phaazon) [reply CMakeLists.txt]: Great! :)

**Comment by KhalilBellakrid** [comment]
I have to handle change of password, I created a JIRA ticket for that : https://github.com/LedgerHQ/lib-ledger-core/pulls
That's why I keep the HODL flag :trollface:

**Comment by phaazon** [comment]
I’m not sure you put the right link :D I’ll have a look in Jira tomorrow, but are you sure you need to implement that feature already (Live / Vault don’t have that feature planned so far, do they?)

**Pull request closed without merging**